src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -104,6 +104,13 @@ public synchronized void reportResults(List<Survey> newSurveys, List<InAppNotifi
             }
         }
 
+        // in the case we do not receive a new variant, this means the A/B test should be turned off
+        if(!hasNewVariants && newVariantsLength == 0) {
```

Comment:
AFAIK we cannot ever have two variants running at once as our SDK does not allow for it... If true, the first case of a new variant such as ```variantA``` or ```variantC``` being applied is handled by the original code above this section. This code is specifically for when the response goes from having variants to having nothing, in which case we would want to remove any active variants as they are no longer activated in the Mixpanel UI.